Pseudocode/Greedy Algorithms/SourceCode.tex

```diff
@@ -0,0 +1,47 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Remove the 'end' at the end of the algorithm
+\usepackage[noend]{algpseudocode}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Activity Selection Problem}
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the Array
+    \Statex
+    \Function{Greedy-Activity-Selector}{$Activity, start, finish$}
+        \State Sort Activities into monotonically increasing order by finish time
```

Comment:
I think it should be `non decreasing order`? Correct me if I am wrong

Reply:
According to Cormen and [Wikipedia](https://en.wikipedia.org/wiki/Activity_selection_problem#Explanation) it should be monotonically increasing order.

Reply:
Oh, I guess that's it then. Please edit it to the original form!

Reply:
But what if the input is `[1,2], [1,2]`?

Reply:
The intended solution is 1, (which is obtained by non-decreasing and not monotonic. What do you think?

Reply:
Don't you think the intended solution will be obtained if sorted monotonically increasing too? I think non-decreasing being a superset will still give the answer but its more accurate to write increasing.

Reply:
But `monotonic` means that if `a` occurs before `b` in sorted order, then `a.finish < b.finish`. However, this isn't the case here. Or am I missing something?

Reply:
Okay, let's keep it `monotonic` for now. I'll try out some examples later and verify them.

Reply:
It's like when we say monotonically increasing if some test case such as yours appears we have to keep one among those with the same finish times. But this raises another question of which among them to select(it should be the one with maximum start time, but the books should clarify)?

Reply:
I think it's better to state that 
> If the `finish` time coincide, then sort by the `start` time

Reply:
And I don't think by monotonically increasing, they mean to remove some entries. (They are only talking about sorting them). But I'm not sure, I'll have to read up the literature.

Reply:
If you want, you can add a comment **after** the end of the algorithm(footer) to verify the statement regarding monotonicity